terraform/03_gke_cluster.tf

```diff
@@ -142,6 +142,43 @@ data "google_compute_default_service_account" "default" {
   ]
 }
 
+# Give service account Observability permissions
+resource "google_project_iam_member" "trace_role" {
+  project = data.google_project.project.project_id
+  role    = "roles/cloudtrace.admin"
+  member  = "serviceAccount:${data.google_compute_default_service_account.default.email}"
+  depends_on = [data.google_compute_default_service_account.default]
+}
+
+resource "google_project_iam_member" "monitoring_role" {
+  project = data.google_project.project.project_id
+  role    = "roles/monitoring.editor"
```

Comment:
If you want to limit permissions as much as possible, could this be `roles/monitoring.metricWriter` or does it need read access for something?
https://cloud.google.com/monitoring/access-control

Reply:
Good catch, I meant to come back to these to tighten the permissions, but must have been distracted by a different PR